components/applications-service/pkg/storage/postgres/services.go

```diff
@@ -364,14 +373,16 @@ func convertComposedServiceToStorage(svc *composedService) *storage.Service {
 //      AND health = 'CRITICAL'     OR health = 'UNKNOWN'
 //      AND environment = 'production'
 //
-func buildWhereConstraintsFromFilters(filters map[string][]string) (string, error) {
+func buildWhereConstraintsFromFilters(filters map[string][]string, first bool) (string, error) {
```

Comment:
My $0.02 here is to avoid modifying this function since it is well isolated and it has full coverage for the construction of `WHERE` statements, I have to viable suggestions on how you could approach this:

1) If you really want a function that appends constraints from an already generated `WHERE` statement, instead of passing a `bool` to the build function, I would suggest creating a new function that wraps this functions by accepting the filters and the previous `WHERE foo=bar` string. Example:
```go
myWhereStatement := "WHERE foo=bar"
err := appendWhereConstraintsFromFilters(filters, myWhereStatement)
if err != nil {
    return err
}
// Here myWhereStatement should have already both filters
```
2) Continue using this function in the same way but organizing the caller function in such way that you can append your extra WHERE statements to the already generated ones. Example: 
```go
func (db *Postgres) GetServicesDistinctValues(fieldName, queryFragment string, filters map[string][]string) ([]string, error) {
	var (
		columnName          = columnNameForField(fieldName)
		fieldNameConstraint = fmt.Sprintf("t.%s ILIKE $1", columnName)
	)

	// maybe create a function that checks any provided filedName?
	if !isValidFilter(fieldName) {
		return nil, errors.Errorf("field name %q is not valid for filtering, valid values are %v", fieldName, validFilterFields)
	}

	whereConstraints, err := buildWhereConstraintsFromFilters(filters)
	if err != nil {
		return nil, err
	}

	// Append to the generated (or not) our fieldName WHERE Constraints
	if len(whereConstraints) == 0 {
		whereConstraints = fmt.Sprintf("WHERE %s", fieldNameConstraint)
	} else {
		whereConstraints = fmt.Sprintf("%s AND %s", whereConstraints, fieldNameConstraint)
	}

	query := fmt.Sprintf("SELECT DISTINCT %[1]s from service_full AS t %[2]s ORDER BY %[1]s ASC LIMIT 100",
		columnNameForField(fieldName),
		whereConstraints,
	)

	matcher := fmt.Sprintf("%s%%", queryFragment)

	var matches []string
	_, err = db.DbMap.Select(&matches, query, matcher)
	if err != nil {
		return nil, err
	}
	return matches, nil
}
```

Reply:
Good suggestion! Thanks.